src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -37,94 +39,216 @@ import GradingActions from './GradingActions';
-    cell: info => <Filterable column={info.column} value={info.getValue()} />
-  }),
-  columnHelper.accessor('groupName', {
-    header: 'Group',
-    cell: info => <Filterable column={info.column} value={info.getValue()} />
-  }),
-  columnHelper.accessor('submissionStatus', {
-    header: 'Progress',
-    cell: info => (
-      <Filterable column={info.column} value={info.getValue()}>
-        <SubmissionStatusBadge status={info.getValue()} />
-      </Filterable>
-    )
-  }),
-  columnHelper.accessor('gradingStatus', {
-    header: 'Grading',
-    cell: info => (
-      <Filterable column={info.column} value={info.getValue()}>
-        <GradingStatusBadge status={info.getValue()} />
-      </Filterable>
-    )
-  }),
-  columnHelper.accessor(({ currentXp, xpBonus, maxXp }) => ({ currentXp, xpBonus, maxXp }), {
-    header: 'Raw XP (+Bonus)',
-    enableColumnFilter: false,
-    cell: info => {
-      const { currentXp, xpBonus, maxXp } = info.getValue();
-      return (
-        <Flex justifyContent="justify-start" spaceX="space-x-2">
-          <Text>
-            {currentXp} (+{xpBonus})
-          </Text>
-          <Text>/</Text>
-          <Text>{maxXp}</Text>
-        </Flex>
-      );
-    }
-  }),
-  columnHelper.accessor(({ submissionId }) => ({ submissionId }), {
-    header: 'Actions',
-    enableColumnFilter: false,
-    cell: info => {
-      const { submissionId } = info.getValue();
-      return <GradingActions submissionId={submissionId} />;
-    }
-  })
-];
-
 type GradingSubmissionTableProps = {
+  totalRows: number;
   submissions: GradingOverview[];
+  // TODO: Abstract pageParams object into a useable type.
+  updateEntries: (
+    group: boolean,
+    pageParams: { offset: number; pageSize: number },
+    filterParams: any
+  ) => void;
 };
 
-const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({ submissions }) => {
-  const dispatch = useDispatch();
+const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({
+  totalRows,
+  submissions,
+  updateEntries
+}) => {
+  // TODO: implement functionality for submission filtering by groups using the group state from useSession.
+
   const tableFilters = useTypedSelector(state => state.workspaces.grading.submissionsTableFilters);
 
   const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([
     ...tableFilters.columnFilters
   ]);
   const [globalFilter, setGlobalFilter] = useState<string | null>(tableFilters.globalFilter);
 
+  const [page, setPage] = useState<number>(0);
+  const [pageSize, setPageSize] = useState<number>(10);
+
+  // This is needed because a filter change is accompanied with a page reset.
+  const resetPage = useCallback(() => setPage(0), [setPage]);
+
+  const [showAllSubmissions, setShowAllSubmissions] = useState(false);
+  const showSubmissionOptions = [
+    { value: false, label: 'ungraded' },
+    { value: true, label: 'all' }
+  ];
+
+  // TODO: implement isAdmin functionality
```

Comment:
Why was this moved from `Grading` into `GradingSubmissionTable`? The logic for filtering by groups does not seem related to the table logic at all? Also, the TODO is still left here.

I'm in favour of bringing it back to the `Grading` component.

Reply:
the reason was to allow users to select their own entries per page, which is tied to the submissions table state.

for grading, it is to simplify the dispatch logic for backend grading parameters since it largely intersects with the filtering by grading status. similarly for grouping, it can largely intersect with filtering by group. noting your comment below, it may not be so easy after all. let me see if i can use your suggestion of prop passing to revert this.

Reply:
Entries per page is fine but in fact, it would be simpler to not move the logic to this table component.

> &hellip; it can largely intersect with filtering by group.

> &hellip; use your suggestion of prop passing

You've already done this; take a look at the type for `updateEntries`: 

```ts
{
  updateEntries: (
    group: boolean,
    pageParams: { offset: number; pageSize: number },
    filterParams: any
  ) => void;
}
```

You are calling it with a `group` parameter, but that's just `limitGroup` which comes from the dropdown. If move the dropdown back to `Grading`, and define `updateGradingOverviewsCallback` like this instead:

```ts
const updateGradingOverviewsCallback = useCallback(
  // No more `group` param
  (pageParams: { offset: number; pageSize: number }, filterParams: Object) => {
    // Replace the `group` argument with `limitGroup` directly
    dispatch(fetchGradingOverviews(limitGroup, pageParams, filterParams));
  },
  [dispatch]
);
```

It saves you the trouble from having to pass the `group` param back and forth, so the type of the `updateEntries` prop can be made simpler.

Reply:
Forgot to update the dependency array in my code suggestion above. Make sure to update to `[dispatch, limitGroup]`, that way, changing `limitGroup` recreates the function, which also triggers a re-render of the grading table thanks to `useEffect`, as it should.